aiidalab_widgets_base/viewers.py

```diff
@@ -123,6 +133,133 @@ def __init__(self, parameter, downloadable=True, **kwargs):
+            style={"description_width": "0px"},
+        )
+        self.size = ipw.FloatText(
+            value=3,
+            layout={"width": "40px"},
+            style={"description_width": "0px"},
+        )
+        self.color = ipw.Dropdown(
+            options=["element", "red", "green", "blue", "yellow", "orange", "purple"],
+            value="element",
+            disabled=False,
+            layout={"width": "80px"},
+            style={"description_width": "initial"},
+        )
+
+        # Delete button.
+        self.delete_button = ipw.Button(
+            description="",
+            icon="trash",
+            button_style="danger",
+            layout={
+                "width": "50px",
+                "visibility": "visible" if deletable else "hidden",
+            },
+        )
+        self.delete_button.on_click(self.delete_myself)
+
+        super().__init__(
+            children=[
+                self.show,
+                self.selection,
+                self.type,
+                self.size,
+                self.color,
+                self.delete_button,
+            ]
+        )
+
+    def delete_myself(self, _):
+        self.master_class.delete_representation(self)
+
+    def sync_myself_to_array_from_atoms_object(self, structure):
+        """Update representation from the structure object."""
+        if structure:
+            if self.uuid in structure.arrays:
+                self.selection.value = list_to_string_range(
+                    np.where(self.atoms_in_representaion(structure))[0], shift=1
+                )
+                return True
+        return False
+
+    def add_myself_to_atoms_object(self, structure):
+        """Add representation array to the structure object. If the array already exists, update it."""
+        if structure:
+            array_representation = np.full(len(structure), -1)
+            selection = np.array(
+                string_range_to_list(self.selection.value, shift=-1)[0], dtype=int
+            )
+            # Only attempt to display the existing atoms.
+            array_representation[selection[selection < len(structure)]] = 1
+            structure.set_array(self.uuid, array_representation)
+            return True
+        return False
+
+    def atoms_in_representaion(self, structure=None):
+        """Return an array of booleans indicating which atoms are present in the representation."""
+        if structure:
+            if self.uuid in structure.arrays:
+                return structure.arrays[self.uuid] >= self.array_threshold
+        return False
+
+    def nglview_parameters(self, indices):
+        """Return the parameters dictionary of a representation."""
+        params = {
+            "type": self.type.value,
+            "params": {
+                "sele": list_to_nglview(indices),
+                "opacity": 1,
+                "color": self.color.value,
+            },
+        }
+        if self.type.value == "ball+stick":
```

Comment:
I am not sure this is possible. The dropdown widget accepts lists, tuples and dictionaries as options.

Reply:
It is. You can provide the options as a list of 2-tuples, where first element is the underlying value (which can be anything, e.g. Enum, and the second is a string that will be displayed to the user). See how I use it here:

https://github.com/ispg-group/aiidalab-ispg/blob/d4ab6ce6972b1be45a6669007cb343c99ecec90c/aiidalab_ispg/app/spectrum.py#L214

imo this makes the code more robust, since you're using a proper type and not just a string, (any misspeling will be caught by the linter for example)

Reply:
> For this, you need to create an Enum that is further converted to a tuple, so why not just create a tuple at first?

Because by using Enum you have a single source of truth, and you can then use it elsewhere when you have code that depends on the options (instead of comparing strings).  In other words using a type is nicer than using strings: https://www.lesinskis.com/stringly-typed-functions.html

But agree, let's not derail this here. :-D